provider/kubernetes/client.go

```diff
@@ -0,0 +1,292 @@
+package kubernetes
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"time"
+
+	"github.com/containous/traefik/old/log"
+	corev1 "k8s.io/api/core/v1"
+	extensionsv1beta1 "k8s.io/api/extensions/v1beta1"
+	kubeerror "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
```

Comment:
I think we are 5-6 versions behind on our k8s client. Can we update to the latest in this PR? we will want to have the latest `client-go` when we start handling CRDs